api/client/lib.py

```diff
@@ -505,12 +505,10 @@ def list_of_series_to_single_series(series_list, add_belongs_to=False, include_h
                 'start_date': point[0],
                 'end_date': point[1],
                 'value': point[2],
-                # list_of_series has unit_id in the series attributes currently. Does
-                # not allow for mixed units in the same series
-                'unit_id': series['series'].get('unitId', None),
+                'unit_id': point[4] if len(point) > 4 else series['series'].get('unitId', None),
                 # input_unit_id and input_unit_scale are deprecated but provided for backwards
                 # compatibility. unit_id should be used instead.
-                'input_unit_id': series['series'].get('unitId', None),
+                'input_unit_id': point[4] if len(point) > 4 else series['series'].get('unitId', None),
```

Comment:
No, see the comment right above:

> input_unit_id and input_unit_scale are deprecated but provided for backwards compatibility. unit_id should be used instead.

In the original versions of the client, we only had `input_unit_id` and `input_unit_scale` since there was no server-side conversion happening at all. When we started doing some conversions, `unit_id` was added to be accurate, but anyone using `input_unit_id` and `input_unit_scale` would've started getting incorrect values since it no longer reflected what unit the value was in.

So those two values are set to the unit_id and 1 so that legacy code will continue to function as expected. It's not documented, and it's discouraged. It's one of the few accumulated cruft things slated for removal in an eventual Major version update, but no timeline on that just yet.